src/vault/client/http.clj

```diff
@@ -136,27 +136,34 @@
     (reset! auth-ref auth-info)))
 
 
-(defn- authenticate-token!
-  "Updates the token ref by storing the given auth token."
-  [client token]
+(defmulti authenticate-type!
```

Comment:
The `-type` suffix doesn't feel right here; I don't think multimethods are typically named after their dispatch. How about `authenticate*` to indicate that it's an internal (ish) helper for the `authenticate!` protocol method?

Reply:
Good call, makes sense to not bind the dispatch key with the name.